drivers/ethernet/eth_nxp_enet.c

```diff
@@ -0,0 +1,1046 @@
+		goto error;
+	}
+
+	if (net_pkt_write(pkt, data->rx_frame_buf, frame_length)) {
+		LOG_ERR("Unable to write frame into the pkt");
+		net_pkt_unref(pkt);
+		k_mutex_unlock(&data->rx_frame_buf_mutex);
+		goto error;
+	}
+
+	k_mutex_unlock(&data->rx_frame_buf_mutex);
+
+#if defined(CONFIG_NET_VLAN)
+	{
+		struct net_eth_hdr *hdr = NET_ETH_HDR(pkt);
+
+		if (ntohs(hdr->type) == NET_ETH_PTYPE_VLAN) {
+			struct net_eth_vlan_hdr *hdr_vlan =
+				(struct net_eth_vlan_hdr *)NET_ETH_HDR(pkt);
+
+			net_pkt_set_vlan_tci(pkt, ntohs(hdr_vlan->vlan.tci));
+			vlan_tag = net_pkt_vlan_tag(pkt);
+
+#if CONFIG_NET_TC_RX_COUNT > 1
+			{
+				enum net_priority prio;
+
+				prio = net_vlan2priority(
+						net_pkt_vlan_priority(pkt));
+				net_pkt_set_priority(pkt, prio);
+			}
+#endif /* CONFIG_NET_TC_RX_COUNT > 1 */
+		}
+	}
+#endif /* CONFIG_NET_VLAN */
+
+	/*
+	 * Use MAC timestamp
+	 */
+#if defined(CONFIG_PTP_CLOCK_NXP_ENET)
+	k_mutex_lock(data->ptp_mutex, K_FOREVER);
+	if (eth_get_ptp_data(get_iface(data, vlan_tag), pkt)) {
+		ENET_Ptp1588GetTimer(config->base, &data->enet_handle,
+					&ptp_time_data);
+		/* If latest timestamp reloads after getting from Rx BD,
+		 * then second - 1 to make sure the actual Rx timestamp is
+		 * accurate
+		 */
+		if (ptp_time_data.nanosecond < ts) {
+			ptp_time_data.second--;
+		}
+
+		pkt->timestamp.nanosecond = ts;
+		pkt->timestamp.second = ptp_time_data.second;
+	} else {
+		/* Invalid value. */
+		pkt->timestamp.nanosecond = UINT32_MAX;
+		pkt->timestamp.second = UINT64_MAX;
+	}
+	k_mutex_unlock(data->ptp_mutex);
+#endif /* CONFIG_PTP_CLOCK_NXP_ENET */
+
+	iface = get_iface(data, vlan_tag);
+#if defined(CONFIG_NET_DSA)
+	iface = dsa_net_recv(iface, &pkt);
+#endif
+	if (net_recv_data(iface, pkt) < 0) {
+		net_pkt_unref(pkt);
+		goto error;
+	}
+
+	return 1;
+flush:
+	/* Flush the current read buffer.  This operation can
+	 * only report failure if there is no frame to flush,
+	 * which cannot happen in this context.
+	 */
+	status = ENET_ReadFrame(config->base, &data->enet_handle, NULL,
+					0, RING_ID, NULL);
+	__ASSERT_NO_MSG(status == kStatus_Success);
+error:
+	eth_stats_update_errors_rx(get_iface(data, vlan_tag));
+	return -EIO;
+}
+
+static void eth_nxp_enet_rx_thread(void *arg1, void *unused1, void *unused2)
+{
+	const struct device *dev = arg1;
+	const struct nxp_enet_mac_config *config = dev->config;
+	struct nxp_enet_mac_data *data = dev->data;
+
+	while (1) {
+		if (k_sem_take(&data->rx_thread_sem, K_FOREVER) == 0) {
+			while (eth_nxp_enet_rx(dev) == 1) {
+				;
+			}
+			/* enable the IRQ for RX */
+			ENET_EnableInterrupts(config->base,
+			  kENET_RxFrameInterrupt | kENET_RxBufferInterrupt);
+		}
+	}
+}
+
+/*
+ ****************************
+ * PHY management functions *
+ ****************************
+ */
+
+static int nxp_enet_phy_reset_and_configure(const struct device *phy)
+{
+	int ret = 0;
+
+	/* Reset the PHY */
+	ret = phy_write(phy, MII_BMCR, MII_BMCR_RESET);
```

Comment:
As per the documentation of a the PHY LAN8720A:

> A Software reset is activated by setting the Soft Reset bit of the Basic Control Register to “1”. All registers bits, except those indicated as “NASR” in the register definitions, are cleared by a Software reset. The Soft Reset bit is self-clearing.
Per the IEEE 802.3u standard, clause 22 (22.2.4.1.1) the reset process will be completed within 0.5s from the setting
of this bit.

I don't think you can configure immediately after this call.
Perhaps we need to add a `reset` function to the PHY API?